numismatic/feeds/bravenewcoin.py

```diff
@@ -0,0 +1,47 @@
+from itertools import product
+import attr
+from .base import Feed
+from ..libs.config import get_config
+
+class BraveNewCoin(Feed):
+    config = get_config()['bravenewcoin'] if 'bravenewcoin' in get_config() \
```

Comment:
Do you want the config to be read at instance creation or class definition time?

If class definition time is ok then we don't need to make it a function `get_config()` and can just read the `config` dict in the `config` module and be done with it.

If you want it to be at instance creation time then this needs to sit inside the `__init__()` method.

Reply:
Sorry, I am not sure how we would get the `config` dict directly without going throught `get_config`... Here is the definition for `get_config`:

```
def get_config():
    for config_file in [Path(os.environ['HOME']) / '.coinrc']:
      config = ConfigParser()
      if config_file.exists():
          config.read(config_file)
          return config

      return None
```

It first reads the config file and then creates the dict.

Another question arises: If you have two accounts for a feed, then you probably need to have two instances of numismatic running. Right now, there would be no elegant way to distinguish between two accounts in config, no matter if it came from the cli or not. I say this because we would ideally need to write an auth utility that can go to the config and parse for multiple accounts. This is all good, however I don't propose doing it in this iteration.

Reply:
With regards to `get_config()`: just remove the `def` in the code you quoted above. Then `config` would exist directly in the namespace and would be importable. Only downside is that it only gets read once when the process is started.

Reply:
With regards to multiple accounts: yes, no way to do that from the config file but easily doable if the credentials are instance state variables. You might not want to run this from the CLI but rather implement in some client code directly.